satellite/metainfo/metainfo.go

```diff
@@ -475,18 +479,30 @@ func (endpoint *Endpoint) filterValidPieces(ctx context.Context, pointer *pb.Poi
 	defer mon.Task()(&ctx)(&err)
 
 	if pointer.Type == pb.Pointer_REMOTE {
-		var remotePieces []*pb.RemotePiece
 		remote := pointer.Remote
+
+		peerIDMap, err := endpoint.mapNodesFor(ctx, remote.RemotePieces)
+		if err != nil {
+			return err
+		}
+
+		var remotePieces []*pb.RemotePiece
 		allSizesValid := true
 		lastPieceSize := int64(0)
 		for _, piece := range remote.RemotePieces {
 
-			// TODO enable piece hash signature verification
+			// Verify storagenode signature on piecehash
+			peerID := peerIDMap[piece.NodeId]
```

Comment:
nit: here we can do `peerID, ok := peerIDMap[piece.NodeId]` and then check `ok` is true or false

Reply:
Done